### PR TITLE
feat: yaml options

### PR DESCRIPTION
**Pull request by urugator**
Types are copied from `yaml` repo:
```
export declare function stringify(value: any, replacer?: Replacer | null, options?: string | number | (DocumentOptions & SchemaOptions & ParseOptions & CreateNodeOptions & ToStringOptions)): string;
```
To keep things simple and in line with `getSpecAsJson` the signature with options only (without replacer) is not supported.

Fixes: https://github.com/metadevpro/openapi3-ts/issues/143

### CODE REVIEW

File: src/dsl/openapi-builder30.ts [hunk 1]
```diff
@@ -45,8 +45,19 @@ export class OpenApiBuilder {
...
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        replacer?: any[] | ((key: any, value: any) => unknown) | null,
+        options?:
+            | string
+            | number
+            | (yaml.DocumentOptions &
+                  yaml.SchemaOptions &
+                  yaml.ParseOptions &
+                  yaml.CreateNodeOptions &
+                  yaml.ToStringOptions)
```
Comment (RobinTail) [inline src/dsl/openapi-builder30.ts:58]: I'd suggest to make a reference to the types of the `yaml.stringify()` parameters instead

Comment (RobinTail) [reply src/dsl/openapi-builder30.ts:58]:
```suggestion
        replacer?: Parameters<typeof yaml.stringify>[1],
        options?: Parameters<typeof yaml.stringify>[2]
```
then you'd not need:
- to reproduce internal typing of another module
- to mute ESLint error

That will also make it future resistant if `yaml` dependency upgrades, @urugator

File: src/dsl/openapi-builder31.ts [hunk 2]
```diff
@@ -45,8 +45,19 @@ export class OpenApiBuilder {
...
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        replacer?: any[] | ((key: any, value: any) => unknown) | null,
+        options?:
+            | string
+            | number
+            | (yaml.DocumentOptions &
+                  yaml.SchemaOptions &
+                  yaml.ParseOptions &
+                  yaml.CreateNodeOptions &
+                  yaml.ToStringOptions)
```
Comment (RobinTail) [inline src/dsl/openapi-builder31.ts:58]: same advice here
https://github.com/metadevpro/openapi3-ts/pull/144/files#r1730896996

File: Changelog.md
Comment (RobinTail) [inline Changelog.md:1]: I'd say it's too much changes to this file that are unrelated to the proposed feature.

**Review by RobinTail — CHANGES_REQUESTED** [review]
suggesting to use `Parameters<typeof yaml.stringify>`

File: src/dsl/openapi-builder30.ts (hunk 1 above)
Comment (urugator) [reply src/dsl/openapi-builder30.ts:58]: I wanted to do that, but it's is not doable for overloaded functions, because TS doesn't know which overload to pick.

File: Changelog.md
Comment (urugator) [reply Changelog.md:1]: It's done by prettier, should I add Changelog.md to prettier ignore?

File: src/dsl/openapi-builder30.ts (hunk 1 above)
Comment (RobinTail) [reply src/dsl/openapi-builder30.ts:58]: pick the one having `replacer`, consider `Extract` for that, @urugator

Comment (urugator) [reply src/dsl/openapi-builder30.ts]: Dunno what you mean with the `Extract`, but turns out `Parameters` are parameters of the last signature, which is the one  we need. So unless they swap or introduce new signatures it should be fine I guess.

**Review by RobinTail — APPROVED** [review]
nice
